packages/manager/apps/telecom/src/app/telecom/pack/slots/exchangeAccount/exchangeAccount.controller.js

```diff
@@ -1,66 +1,93 @@
+  constructor(
+    $scope,
+    $http,
+    $q,
+    $stateParams,
+    coreURLBuilder,
+    OvhApiPackXdslExchangeAccount,
+    iceberg,
+  ) {
+    this.$scope = $scope;
+    this.$http = $http;
+    this.$q = $q;
+    this.$stateParams = $stateParams;
+    this.coreURLBuilder = coreURLBuilder;
+    this.OvhApiPackXdslExchangeAccount = OvhApiPackXdslExchangeAccount;
+    this.iceberg = iceberg;
+  }
 
-    $scope.loaders = {
+  $onInit() {
+    this.$scope.loaders = {
       services: true,
     };
+    return this.getExchangeAccounts()
+      .then((services) => {
+        this.services = services;
+      })
+      .finally(() => {
+        this.$scope.loaders.services = false;
+      });
+  }
 
-    return OvhApiPackXdslExchangeAccount.Services()
+  getExchangeAccountServices() {
+    return this.OvhApiPackXdslExchangeAccount.Services()
       .v6()
       .query({
-        packName: $stateParams.packName,
+        packName: this.$stateParams.packName,
       })
       .$promise.then((serviceIds) =>
-        $http
-          .get('/email/exchange/*/service/*/account?$aggreg=1', {
-            serviceType: 'apiv7',
-          })
-          .then((servicesParam) => {
-            let services = servicesParam;
-            services = get(services, 'data');
-            self.services = filter(
-              map(
-                filter(services, (service) => service.value !== null),
-                (service) => {
-                  const splittedPath = service.path.split('/');
-                  return angular.extend(service.value, {
-                    organizationName: splittedPath[3],
-                    exchangeService: splittedPath[5],
-                    managerUrl: coreURLBuilder.buildURL(
-                      'exchange',
-                      '#/configuration/exchange_hosted/:organization/:productId',
-                      {
-                        organization: splittedPath[3],
-                        productId: splittedPath[5],
-                        tab: 'ACCOUNT',
-                      },
-                    ),
-                  });
-                },
-              ),
-              (service) =>
-                serviceIds.indexOf(`${service.exchangeService}-${service.id}`) >
-                -1,
-            );
-          }),
-      )
-      .finally(() => {
-        $scope.loaders.services = false;
-      });
+        serviceIds.map((serviceId) => {
+          const regex = /^(.*)-(.*)$/gi;
+          const [, exchangeService, accountId] = regex.exec(serviceId);
+          return { exchangeService, accountId };
+        }),
+      );
   }
 
-  init();
+  getExchangeAccounts() {
+    return this.getExchangeAccountServices()
+      .then((exchangeServicesIds) => {
+        const accountIdsByExchangeService = exchangeServicesIds.reduce(
+          (all, { exchangeService, accountId }) => {
+            return {
+              ...all,
+              [exchangeService]: [...(all[exchangeService] || []), accountId],
```

Comment:
This line is a bit hard to understand, a comment would me nice to improve maintainability